Assets/Scripts/Layers/Adapters/DataTypeAdapters/GeoJSONImportAdapter.cs

```diff
@@ -63,21 +65,26 @@ private void CreateGeoJSONLayer(LocalFile localFile, UnityEvent<string> onErrorC
             if(localFile.SourceUrl.Length > 0)
                 geoJsonLayerName = localFile.SourceUrl;    
         
-            //Create a new geojson layer with random color (untill UI provides ways to choose colors)
             GeoJsonLayerGameObject newLayer = Instantiate(layerPrefab);
             newLayer.Name = geoJsonLayerName;
             newLayer.gameObject.name = geoJsonLayerName;
             if (onErrorCallback != null)
                 newLayer.OnParseError.AddListener(onErrorCallback.Invoke);
 
+            //GeoJSON layer+visual colors are set to random colors until user can pick colors in UI
+            var randomLayerColor = Color.HSVToRGB(UnityEngine.Random.value, UnityEngine.Random.Range(0.5f, 1f), 1);
+            randomLayerColor.a = 0.5f;
+            newLayer.LayerData.Color = randomLayerColor;
+            
+            var symbolizer = newLayer.LayerData.DefaultSymbolizer;
+            symbolizer?.SetFillColor(randomLayerColor);
+            symbolizer?.SetStrokeColor(randomLayerColor);
+            
             var localPath = localFile.LocalFilePath;
-            var fileName = Path.GetFileName(localPath);
             var propertyData = newLayer.PropertyData as LayerURLPropertyData;
-
-            if (localFile.SourceUrl.StartsWith("http"))
-                propertyData.Data = AssetUriFactory.CreateRemoteAssetUri(localFile.SourceUrl);
-            else
-                propertyData.Data = AssetUriFactory.CreateProjectAssetUri(localPath);
+            propertyData.Data = localFile.SourceUrl.StartsWith("http") 
```

Comment:
is het een idee om localFile een IsRemote functie/property te geven die je kan opvragen?

Reply:
Het zou beter zijn om van SourceUrl een URI te maken, dan kan je daar IsRemoteFile() op aanroepen (is een Uri extensie die bestaat); maar dat ook aanpassen vond ik een te grote scope increase :D